src/Agent.Worker/ExecutionContext.cs

```diff
@@ -621,10 +621,25 @@ private void JobServerQueueThrottling_EventReceived(object sender, ThrottlingEve
                     uriBuilder.Path += (Variables.System_TFCollectionUrl.EndsWith("/") ? "" : "/") + "_usersSettings/usage";
                     query["tab"] = "pipelines";
                     query["queryDate"] = queryDate;
-
+                    
+                    // Global RU link
                     uriBuilder.Query = query.ToString();
-
-                    this.Warning(StringUtil.Loc("ServerTarpitUrl", uriBuilder.ToString()));
+                    string global = StringUtil.Loc("ServerTarpitUrl", uriBuilder.ToString());
+
+                    if (!String.IsNullOrEmpty(this.Variables.Build_DefinitionName))
+                    {
+                        query["keywords"] = this.Variables.Build_Number;
+                        query["definition"] = this.Variables.Build_DefinitionName;
+                    }
+                    else if (!String.IsNullOrEmpty(this.Variables.Release_ReleaseName))
+                    {
+                        query["keywords"] = this.Variables.Release_ReleaseId;
+                        query["definition"] = this.Variables.Release_ReleaseName;
+                    }
+
+                    // RU link scoped for the build/release
+                    uriBuilder.Query = query.ToString();
+                    this.Warning($"{global}\n{StringUtil.Loc("ServerTarpitUrlScoped", uriBuilder.ToString())}");
```

Comment:
should we just do this?
```C#
this.Warning(global);
this.Warning(StringUtil.Loc("ServerTarpitUrlScoped", uriBuilder.ToString()));
```

Reply:
I've no strong opinions either way. The only functional difference will be four total warning messages vs. three, which the pipelines UI already groups them nicely.

Example:
![image](https://user-images.githubusercontent.com/2474119/46894776-a6b7e380-ce43-11e8-9f46-eafd16e93a75.png)